Code/Java/CircularQueue

```diff
@@ -0,0 +1,402 @@
+    int cq[];
+    int first, last;
+
+    CircularQueue(int size)
+    { 
+        //This is a default constructor to declare Circular Queue and global parameters first and last.
+        cq = new int[size];
+        first =-1;
+        last = -1;
+    }
+
+    public void insert(int value)
+    {
+        /**
+         * This function is used to insert an element into the circular queue. 
+         * In a circular queue, the new element is always inserted at last position. 
+         * Steps:
+         * 1. Check if Circular Queue cq is full (first==0 && last==cq.length-1 || first==last+1).
+         * 2. If it is full then display "Overflow" and return to main function. If Circular Queue is not full then go to Step 3. 
+         * 3. Check if (first==-1).
+         *    If it is, then set first=last=0.
+         *    Else check if (last==cq.length-1).
+         *    If it is, then set last=0.
+         *    Else increment last by 1.
+         * 4. Insert value at position last.
+         */
+        if(first==0 && last==cq.length-1 || first==last+1)
+        { 
+            System.out.println("Overflow"); return;
+        }
+        else if(first==-1)
+        {
+            first=last=0;
+        }
+        else if(last==cq.length-1)
+        {
+            last=0;
+        }
+        else
+        {
+            last++;
+        }
+        cq[last]=value;
+    }
+
+    public int delete()
+    {
+        /**
+         * This function is used to delete an element from the Circular Queue. 
+         * In a circular queue, the element is always deleted from first position. 
+         * Steps:
+         * 1. Check if the Circular Queue cq is empty (first==-1).
+         * 2. If it is empty then display "Underflow" and return 0. If the Circular Queue is not empty then go to Step 3.
+         * 3. Check if (first==last).
+         *    If it is, then call reset function [first=last=-1].
+         *    Else check if (first==cq.length-1).
+         *    If it is, then set first=0.
+         *    Else increment first by 1.
+         * 4. Return the element at position first.
+         */
+
+        if(first==-1)
+        {
+            System.out.println("Underflow"); return 0; 
+        }
+        else
+        {
+            int temp = cq[first];
+            if(first==last)
+            { 
+                reset(); 
+            }
+            else if(first==cq.length-1)
+            {
+                first=0;
+            }
+            else
+            {
+                first++;
+            }
+            return temp;
+        }
+    }
+
+    private void display()
+    {
+        /**
+         * This function is used to display a Circular Queue.
+         * Steps:
+         * 1. Check if Circular Queue is empty (first==-1)
+         * 2. If it is empty then display "Circular Queue is Empty". If the Circular Queue is not empty then go to Step 3.
+         * 3. Start for loop from the position pointed by variable first, incrementing i by 1.
```

Comment:
Add these in form of single line comments in between the code wherever required, Only add necessary comments.

Reply:
Made the changes.